api/app/utils/redapp.py

```diff
@@ -0,0 +1,316 @@
+        # 'S1': - no mapping
+        # 'S2': - no mapping
+        # 'S3': - no mapping
+    }
+    if fuel_type in fuel_type_map:
+        return fuel_type_map[fuel_type]
+    raise UnmappedFuelType(fuel_type)
+
+
+def FBPCalculateStatisticsCOM(elevation: float,  # pylint: disable=invalid-name, too-many-arguments
+                              latitude: float,
+                              longitude: float,
+                              time_of_interest: datetime,
+                              fuel_type: str,
+                              ffmc: float,
+                              dmc: float,
+                              dc: float,
+                              bui: float,
+                              wind_speed: float,
+                              wind_direction: float,
+                              percentage_conifer: float,
+                              percentage_dead_balsam_fir: float,
+                              grass_cure: float,
+                              crown_base_height: float):
+    """ Uses method FBPCalculateStatisticsCOM on class ca.cwfgm.fbp.FBPCalculations to calculate
+    fire behaviour.
+
+    public class FBPCalculations {
+
+        # methods:
+        public FBPCalculations() {
+        initializeInputValues();
+        initializeOutputValues();
+        }
+        public void FBPCalculateStatisticsCOM()
+    }
+    """
+    # pylint: disable=too-many-locals
+    # we have to do a late import, otherwise we get a segmentation fault.
+    import jnius  # pylint: disable=import-outside-toplevel
+    try:
+        Calendar = jnius.autoclass('java.util.Calendar')  # pylint: disable=invalid-name
+        TimeZone = jnius.autoclass('java.util.TimeZone')  # pylint: disable=invalid-name
+        FBPCalculationsClass = jnius.autoclass('ca.cwfgm.fbp.FBPCalculations')  # pylint: disable=invalid-name
+
+        if percentage_dead_balsam_fir is None:
+            percentage_dead_balsam_fir = 0.0
+        if grass_cure is None:
+            grass_cure = 0.0
+
+        fbp = FBPCalculationsClass()
+        fbp.elevation = elevation
+        fbp.latitude = latitude
+        fbp.longitude = longitude
+        fbp.ffmc = ffmc
+        fbp.dmc = dmc
+        fbp.dc = dc
+        fbp.useBui = True
+        fbp.useBuildup = True
+        fbp.bui = bui
+        fbp.windSpeed = wind_speed
+        fbp.windDirection = wind_direction
+        fbp.fuelType = red_app_fuel_type_map(fuel_type)
+        fbp.conifMixedWood = percentage_conifer
+        fbp.deadBalsam = percentage_dead_balsam_fir
+        fbp.grassCuring = grass_cure
+        fbp.crownBase = crown_base_height
+
+        fbp.useSlope = False
+
+        gmt = Calendar.getInstance(TimeZone.getTimeZone("GMT"))
+        gmt.set(time_of_interest.year, time_of_interest.month, time_of_interest.day,
+                time_of_interest.hour, time_of_interest.minute, time_of_interest.second)
+        fbp.m_date = gmt
```

Comment:
Does redapp expect gmt time?

Reply:
It takes timezone into account, so you just need to be sure to set one.
`
this.fbpCalculations.m_date.setTimeZone(
        TimeZone.getTimeZone(this.app.getSelectedTimeZone().toString()
          .substring(0, i - 1)));
`
but really.... they throw away all of that, and just use the day of the year:
`
int julian = (int)newDate.getDayOfYear(1L);
`

Reply:
Can we use a PST timezone just to avoid any confusion?

Reply:
Neither - will address on another branch and have it take the TZ from the input instead of specifying it here.